Automattic/wp-calypso
# Coming Soon: ensure consistency across settings pages and site badge for v1 and v2

## Changes proposed in this Pull Request

This PR ensures that the settings pages and site badges bar appear in Coming Soon mode regardless of whether a site is still on v1 or has a new site on v2.

## Testing instructions

Take a deep breath.

When testing, keep in mind that changing settings calypso.localhost:3000 will using Coming Soon v2 mode.

To switch back to private, Coming Soon mode you can visit the same site's setting in wordpress.com.

### How to test Coming Soon v2

Create a Coming Soon v2 site over at calypso.localhost:3000/new or calypso.localhost:3000/start/domains

Check the settings of your new site 

![image](https://user-images.githubusercontent.com/6458278/99029187-f7234180-25c5-11eb-9f1f-5858ec51720a.png)

Check the sidebar site badge

![image](https://user-images.githubusercontent.com/6458278/99029196-fc808c00-25c5-11eb-83ca-fa00acf81fc0.png)

In a logged-out state (incognito browser for example), check that the Coming Soon page is shown:

![image](https://user-images.githubusercontent.com/6458278/99029239-1f12a500-25c6-11eb-8fdb-73940be249a5.png)

Upgrade to a Business account and install a plugin to kick off Atomic transfer.

Check the settings page again. The setting should have remained unchanged.

Launch the site. Check the settings page again.

![image](https://user-images.githubusercontent.com/6458278/99029275-30f44800-25c6-11eb-9d69-d7365f9c5a23.png)

In a logged-out state, check that the site is live and public.

Toggle Coming Soon mode. Check that the Coming Soon page is displayed/hidden accordingly.

### How to test Coming Soon v2 for existing sites 

You can create Coming Soon v1 sites on the wordpress.com domain, e.g. wordpress.com/new or wordpress.com/start/domains. 

Once you've done that, load up the created site in Calypso using calypso.localhost:3000/settings/general/{your_site}

#### On an existing, unlaunched site

Check the settings:

![image](https://user-images.githubusercontent.com/6458278/99029387-731d8980-25c6-11eb-9269-3cc8a30d9d99.png)

Check the sidebar site badge:

![image](https://user-images.githubusercontent.com/6458278/99029397-7add2e00-25c6-11eb-8954-f23a27ad96da.png)

Launch then toggle back to coming soon mode. 

You should be in public coming soon mode.

```
wp option --url=v1comingsoontesttodayqwe123.wordpress.com get wpcom_public_coming_soon === 1

wp option --url=v1comingsoontesttodayqwe123.wordpress.com get wpcom_coming_soon === 0

wp option --url=v1comingsoontesttodayqwe123.wordpress.com get blog_public === 0
```

#### On an existing, launched site in Coming Soon v1 mode

In an existing site that you've launched and toggled back to Coming soon (private v1) mode.

Hint: you can toggle back to Coming soon (private v1) mode by switching over to wordpress.com/settings/general/{your_site}, then returning to wpcalypso.wordpress.com/settings/general/{your_site}.

Check that the settings page show's coming soon:

![image](https://user-images.githubusercontent.com/6458278/99029460-9ea07400-25c6-11eb-99a6-77b5ad32134b.png)

Check the options:

```
wp option --url=v1comingsoontesttodayqwe123.wordpress.com get wpcom_public_coming_soon === 0

wp option --url=v1comingsoontesttodayqwe123.wordpress.com get wpcom_coming_soon === 1

wp option --url=v1comingsoontesttodayqwe123.wordpress.com get blog_public === -1
```

Toggle coming soon mode and check the options

```
wp option --url=v1comingsoontesttodayqwe123.wordpress.com get wpcom_public_coming_soon === 1

wp option --url=v1comingsoontesttodayqwe123.wordpress.com get wpcom_coming_soon === 0

wp option --url=v1comingsoontesttodayqwe123.wordpress.com get blog_public === 0
```

### On an existing Atomic site

For unlaunched Atomic sites, check the settings

![image](https://user-images.githubusercontent.com/6458278/99029520-c8599b00-25c6-11eb-8b88-35961d8c2c09.png)

For launched Atomic sites, check the settings

![image](https://user-images.githubusercontent.com/6458278/99029534-d14a6c80-25c6-11eb-84bc-3429b2687619.png)

Toggling Coming Soon should work as expected.
